code/iLasso_for_SCINGE.m

```diff
@@ -68,40 +69,25 @@
 tval(tind) = [];
 B = sum(ttime<=(L*Dt));
 N1 = size(ttime, 2);
-
+remind = find(~Xdrop(pa(1),:)&ptime>L*Dt);
 % Build the matrix elements
 Am = (zeros(N1-B, numregs*L));
+X = X(pa,:).*rind(pa,:);
 % Building the design matrix
 for k = 1:L
     Kp2 = m.fullKp(1,k);
-    for j = 1:numregs
-        rind = find(Xdrop(pa(j),:));
-        ySelect = (full(X(pa(j),:)'));
-        refL = length(ySelect);
-        tSelect = ptime;
-        ySelect(rind) = [];
-        tSelect(rind) = [];
-        refind = 1:refL;
-        remind = refind(~(ismember(refind,tind))); % setdiff(refind,tind);
-        remind = union(1:remind(B+1)-1,tind);
-        rind = refind(~(ismember(refind,rind)));
-        remind = refind(~(ismember(refind,remind)));
-           switch krnl
```

Comment:
We only support one kernel now so we do not need the argument in `iLasso_for_SCINGE(m, outs, lambda, krnl,L,dDt,SIG,params)` and can also remove the function documentation about the `krnl` argument.